packages/react-native-editor/__device-tests__/pages/editor-page.js

```diff
@@ -45,6 +45,25 @@ class EditorPage {
 		return await this.driver.hasElementByAccessibilityId( 'block-list' );
 	}
 
+	// For text blocks, e.g Paragraph, Heading
+	async getTextBlockLocatorAtPosition( blockName, position = 1 ) {
+		// iOS needs a click before
+		if ( ! isAndroid() ) {
+			const textBlockLocator = `(//XCUIElementTypeButton[contains(@name, "${ blockName } Block. Row ${ position }")])`;
+			const textBlock = await waitForVisible(
+				this.driver,
+				textBlockLocator
+			);
+			await textBlock.click();
+		}
```

Comment:
For iOS, it looks like we would still require a click to get the text value, if we don't do it as part of the function, we would need to call it before getting the text value. The reason is how the locators are structured:

In this example, when the first paragraph is not the active one, we wouldn't be able to get the text value of the element as the text element will only be visible when the block is selected:
<img width="1274" alt="image" src="https://user-images.githubusercontent.com/17252150/165012673-ddb6d7d8-38d0-4295-8f2a-501bfb0504f6.png">

When it is selected (after clicking on it), there is a new `XCUIElementTypeTextView` locator visible where we would be able to get the text value using `.text()`:
<img width="1273" alt="image" src="https://user-images.githubusercontent.com/17252150/165012865-b3cad072-026f-4569-b4fd-f9d29a87d8d8.png">

The previous implementation also has a click, however, it wasn't that obvious as it was hidden in another function, previously this was the flow to get the text value:

From the test file, call [getTextForParagraphBlockAtPosition()](https://github.com/WordPress/gutenberg/blob/60bddd382c2b66c1c9ccd726717087a376958fa2/packages/react-native-editor/__device-tests__/pages/editor-page.js#L540-L554) that would look like this, notice that there is a `await block.click();` action to click the block:
```
	async getTextForParagraphBlockAtPosition( position ) {
		// Select block first.
		let block = await this.getBlockAtPosition(
			this.paragraphBlockName,
			position
		);
		await block.click();

		block = await this.getBlockAtPosition(
			this.paragraphBlockName,
			position
		);
		const text = await this.getTextForParagraphBlock( block );
		return text.toString();
	}
```

The above would then need to call [getTextForParagraphBlock()](https://github.com/WordPress/gutenberg/blob/60bddd382c2b66c1c9ccd726717087a376958fa2/packages/react-native-editor/__device-tests__/pages/editor-page.js#L532-L538):
```
	async getTextForParagraphBlock( block ) {
		const textViewElement = await this.getTextViewForParagraphBlock(
			block
		);
		const text = await textViewElement.text();
		return text.toString();
	}
```

Which would need to call [getTextViewForParagraphBlock()](https://github.com/WordPress/gutenberg/blob/60bddd382c2b66c1c9ccd726717087a376958fa2/packages/react-native-editor/__device-tests__/pages/editor-page.js#L488-L501) to get the text value:
```
	async getTextViewForParagraphBlock( block ) {
		let textViewElementName = 'XCUIElementTypeTextView';
		if ( isAndroid() ) {
			textViewElementName = 'android.widget.EditText';
		}

		const accessibilityId = await block.getAttribute(
			this.accessibilityIdKey
		);
		const blockLocator = `//*[@${
			this.accessibilityIdXPathAttrib
		}=${ JSON.stringify( accessibilityId ) }]//${ textViewElementName }`;
		return await this.driver.elementByXPath( blockLocator );
	}
```

With this change, we can remove both `getTextForParagraphBlock()` and `getTextViewForParagraphBlock()` and the click on `getTextForParagraphBlockAtPosition()` as it's not required for Android.

Reply:
It's curious that the text element can't be queried until it's selected, but I guess this might be related to how it's implemented on the native side 🤔 .

Thank you very much @jostnes for the clarification 🙇, I'll go ahead and mark the comment as resolved.